vm/src/hint_processor/builtin_hint_processor/ec_utils.rs

```diff
@@ -144,10 +144,10 @@ pub fn recover_y_hint(
     let p_x = get_integer_from_var_name("x", vm, ids_data, ap_tracking)?.into_owned();
     let p_addr = get_relocatable_from_var_name("p", vm, ids_data, ap_tracking)?;
     vm.insert_value(p_addr, p_x)?;
-    let p_y = biguint_to_felt(
-        &recover_y(&felt_to_biguint(p_x))
+    let p_y = Felt252::from(
+        &recover_y(&p_x.to_biguint())
```

Comment:
I see some redundant conversion here. Maybe `recover_y` could simply return the `Felt252` instead.

Reply:
`recover_y` is also used by other functions that use BigUint and don't need to convert  from Felt to use it, so it is better to leave the conversion outside and let the function just handle BigUint